src/cppcache/src/statistics/AtomicStatisticsImpl.cpp

```diff
@@ -330,7 +329,13 @@ double AtomicStatisticsImpl::_incDouble(int32_t offset, double delta) {
     throw IllegalArgumentException(s);
   }
 
-  return (doubleStorage[offset] += delta);
+  double expected = doubleStorage[offset];
+  double value;
+  do {
+    value = expected + delta;
+  } while (!doubleStorage[offset].compare_exchange_weak(expected, value));
```

Comment:
Yes. Some of the documentation for std::atomic suck out there and don't clearly state that expected is updated if the compare fails.